lib/backend/compat/compat.go

```diff
@@ -197,22 +197,22 @@ func (c *ModelAdaptor) Apply(d *model.KVPair) (*model.KVPair, error) {
 }
 
 // Delete an entry in the datastore.  This errors if the entry does not exists.
-func (c *ModelAdaptor) Delete(ctx context.Context, k model.Key, rev string) error {
+func (c *ModelAdaptor) Delete(ctx context.Context, k model.Key, rev string) (*model.KVPair, error) {
 	var err error
 	switch key := k.(type) {
 	case model.NodeKey:
 		p, o := toNodeDeleteComponents(key)
 		if err = c.applyOrDeleteSubcomponents(ctx, o); err != nil {
-			return err
+			return nil, err
 		}
-		if err = c.client.Delete(ctx, p.Key, rev); err != nil {
-			return err
+		if _, err = c.client.Delete(ctx, p.Key, rev); err != nil {
+			return nil, err
 		}
-		return nil
+		return nil, nil
```

Comment:
shouldn't you be returning `KVPair` and `nil` here?

Reply:
yeah - but we aren't using the compat module anymore.   I'm keeping it around in case I copy some of the code, at the moment I just want it to compile.